src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -35,17 +36,21 @@ private void ExecuteForegroundAction(Action action)
             {
                 lock (_queueGate)
                 {
-                    _foregroundTaskQueue = _foregroundTaskQueue.SafeContinueWith(
-                        _ =>
+                    _foregroundTaskQueue = _foregroundTaskQueue.SafeContinueWithFromAsync(
```

Comment:
This pattern can create alot of intermediate tasks...is there a nicer way?

Reply:
➡️ Will file a follow-up issue and mark as higher priority due to potential perf impact. We will have multiple alternatives to pick from.